aptos-move/aptos-gas/src/transaction.rs

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+use crate::gas_meter::{FromOnChainGasSchedule, InitialGasSchedule, ToOnChainGasSchedule};
+use std::collections::BTreeMap;
+
+macro_rules! define_gas_parameters_for_transaction {
+    ($([$name: ident, $key: literal, $initial: expr $(,)?]),* $(,)?) => {
+        /// Transaction-level gas parameters.
+        ///
+        /// Note: due to performance considerations, this is represented as a fixed struct instead of
+        /// some other data structures that require complex lookups.
+        #[derive(Debug, Clone)]
+        pub struct TransactionGasParameters {
+            $(pub $name : u64),*
+        }
+
+        impl FromOnChainGasSchedule for TransactionGasParameters {
+            fn from_on_chain_gas_schedule(gas_schedule: &BTreeMap<String, u64>) -> Option<Self> {
+                Some(Self { $($name: gas_schedule.get(&format!("txn.{}", $key)).cloned()?),* })
+            }
+        }
+
+        impl ToOnChainGasSchedule for TransactionGasParameters {
+            fn to_on_chain_gas_schedule(&self) -> Vec<(String, u64)> {
+                vec![$((format!("txn.{}", $key), self.$name)),*]
+            }
+        }
+
+        impl TransactionGasParameters {
+            pub fn zeros() -> Self {
+                Self {
+                    $($name: 0),*
+                }
+            }
+        }
+
+        impl InitialGasSchedule for TransactionGasParameters {
+            fn initial() -> Self {
+                Self {
+                    $($name: $initial),*
+                }
+            }
+        }
+
+        #[test]
+        fn keys_should_be_unique() {
+            let mut map: BTreeMap<&str, ()> = BTreeMap::new();
+
+            for key in [$($key),*] {
+                assert!(map.insert(key, ()).is_none());
+            }
+        }
+    }
+}
+
+define_gas_parameters_for_transaction!(
+    // The flat minimum amount of gas required for any transaction.
+    // Charged at the start of execution.
+    [min_transaction_gas_units, "min_transaction_gas_units", 600],
+    // Any transaction over this size will be charged an additional amount per byte.
+    [large_transaction_cutoff, "large_transaction_cutoff", 600],
+    // The units of gas that to be charged per byte over the `large_transaction_cutoff` in addition to
+    // `min_transaction_gas_units` for transactions whose size exceeds `large_transaction_cutoff`.
+    [intrinsic_gas_per_byte, "intrinsic_gas_per_byte", 8],
+    // ~5 microseconds should equal one unit of computational gas. We bound the maximum
+    // computational time of any given transaction at roughly 20 seconds. We want this number and
+    // `MAX_PRICE_PER_GAS_UNIT` to always satisfy the inequality that
+    // MAXIMUM_NUMBER_OF_GAS_UNITS * MAX_PRICE_PER_GAS_UNIT < min(u64::MAX, GasUnits<GasCarrier>::MAX)
+    // NB: The bound is set quite high since custom scripts aren't allowed except from predefined
+    // and vetted senders.
```

Comment:
This isn't true.

Reply:
It's an old comment I copied over. Will get it deleted